rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -367,13 +370,41 @@ module Node {
     private CallExprBaseCfgNode call_;
     private RustDataFlow::ArgumentPosition pos_;
 
-    ExprArgumentNode() { isArgumentForCall(n, call_, pos_) }
+    ExprArgumentNode() {
+      isArgumentForCall(n, call_, pos_) and
+      // For receivers in method calls the `ReceiverNode` is the argument.
+      not call_.(MethodCallExprCfgNode).getReceiver() = n
```

Comment:
Perhaps instead remove the `arg = call.(MethodCallExprCfgNode).getReceiver() and pos.isSelf()` case inside `isArgumentForCall`, and then remove this restriction and add a `e = any(MethodCallExprCfgNode mc).getReceiver()` case to `TExprPostUpdateNode`.